src/pytorch_lightning/strategies/strategy.py

```diff
@@ -136,7 +136,7 @@ def setup_optimizers(self, trainer: "pl.Trainer") -> None:
         Args:
             trainer: the Trainer, these optimizers should be connected to
         """
-        if trainer.state.fn not in (TrainerFn.FITTING, TrainerFn.TUNING):
+        if trainer.state.fn != TrainerFn.FITTING:
```

Comment:
there's no way since the TrainerFn is overridden by their specific call.. fit/validate/test/predict.
I was going to propose to deprecate and remove `trainer.tune` in future since we have callbacks now but since `TrainerFn.TUNING` is redundant completely, I think deprecating it in this release would be fine.